pwnlib/util/misc.py

```diff
@@ -280,10 +280,13 @@ def run_in_new_terminal(command, terminal=None, args=None, kill_at_exit=True, pr
         pid = p.pid
 
     if kill_at_exit:
-        if terminal == 'tmux':
-            atexit.register(lambda: os.kill(pid, signal.SIGTERM))
-        else:
-            atexit.register(lambda: p.terminate())
```

Comment:
Does this not change semantics in a bad way?

Reply:
`p.terminate()` just does a `SIGTERM` if you dig into the code.

`pid = p.pid` so it's effectively `os.kill(p.pid, SIGTERM)` which is what `.terminate()` does.

This just makes it take a common code path, since we don't need two separate ways to kill processes at exit.

Reply:
But it keeps the process reference alive, which can also handle SIGCHLD iirc, and prevent us from killing/wait()ing the same pid twice, and p.terminate() is more portable (not that we care about non-posix OSes anyway).

Reply:
The net effect is the same, we don't update any internal state on `p.terminate()` -- it literally just boils down to `subprocess.Popen.terminate` doing the same thing:

```
        def terminate(self):
            """Terminate the process with SIGTERM
            """
            self.send_signal(signal.SIGTERM)
```

```py
        def send_signal(self, sig):
            """Send a signal to the process."""
            # Skip signalling a process that we know has already died.
            if self.returncode is None:
                os.kill(self.pid, sig)
```